sound/soc/sof/ipc4-pcm.c

```diff
@@ -216,6 +281,21 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	if (!pipeline_list->pipelines)
 		return 0;
 
+	/*
+	 * Every pipeline of a chained DMA stream should have use_chain_dma attribute, so
+	 * this loop is only necessary for checking the correctness of the topology. The loop
+	 * together with the check in sof_ipc4_chain_dma_trigger() completes the check.
+	 */
+	for (i = pipeline_list->count - 1; i >= 0; i--) {
+		spipe = pipeline_list->pipelines[i];
+		pipe_widget = spipe->pipe_widget;
+		pipeline = pipe_widget->private;
+
+		/* If a pipeline in the list uses chained DMA, trigger chained DMA and return */
+		if (pipeline->use_chain_dma)
+			return sof_ipc4_chain_dma_trigger(sdev, pipeline_list, state);
```

Comment:
I find this loop a bit confusing.

The comments talk about "every pipeline of a chained DMA" and "if a pipeline uses chained DMA", but in practice the code triggers the FIRST pipeline that's found.

We should clarify that for chain DMA this is intentional, the pipelines are notional and using the first is just fine.

Reply:
Added more comments to this place and before sof_ipc4_chain_dma_trigger().